src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java

```diff
@@ -291,11 +291,6 @@ public PollingResult compareRemoteRevisionWith(Job<?, ?> job, Launcher launcher,
 		PollingResult state = PollingResult.NO_CHANGES;
 		Node node = NodeHelper.workspaceToNode(buildWorkspace);
 
-		if (job.isBuilding() && !isConcurrentBuild(job)) {
```

Comment:
I am not sure removing this check is the right change, by removing this test you are guaranteed to increasing the load on the Perforce server. Naturally, the more polling jobs you have the high load you put on the server.  

I think a better change would be to have a toggle that allows polling during a build. That way you get the functionality you want without forcing other to dramatically increase the load on their Perforce server.  so something like:
`if (((job.isBuilding() && !isConcurrentBuild(job)) || !allowPollingWhenJobRunning)`

I understand checking for changes isn't exactly an intensive operation but still.

Reply:
The condition isConcurrentBuild was added in change 23218 for JENKINS-34052.  The original isBuilding was added wayback in 14040.  Removing the code will have side effects, but I do not have test coverage for that.

Why do you want a polling result for non concurrent builds when a build is in progress?  Perhaps we could remove the changes from the in progress build when calculating the polling?

Reply:
Even if concurrent builds are disabled, allowing polling to still occur will cause any new changes to queue up a new build. This reduces/removes the quiet period in between builds that is at most your polling interval.

Reply:
@peanderson2016 That would probably be a more robust solution. If @p4paul thinks that is a good approach, I can look into that. Would this be a global setting, or a per-job/SCM configuration basis?

Reply:
I think its best to have it as a per-job/SCM configuration. This way you can have it on the exact job your want and nothing else. Also, with JobDSL its not a huge pain for people to enable it for all jobs if that is what they want.

Reply:
I'm going to need to write a test as I'm concerned that if we remove the isBuilding check you will get a polling result for the build in progress.  

For example, if change 200 was building and you poll, the plugin would report that 200 needs building and you would schedule another build for 200 every polling interval.

The private method pollWorkspace takes a lastRun argument, supplied by `job.getLastBuild();`.  The last run would normally be the last completed build, not the build inprogress.

Reply:
I have submitted a test case to verify the polling behaviour during a build and it looks to pass with or without the proposed change.  
I see no need to make this a configuration option and happy to remove the check as originally proposed in the pull request.